practice-two/src/components/FormInput/FormInput.jsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from 'react';
+import './FormInput.css';
+import PropTypes from 'prop-types';
+import { MESSAGES, REGEX } from '@constants';
+
+const FormInput = (props) => {
+  const [focused, setFocused] = useState(false);
+  const { label, errorMessage, onChange, ...inputProps } = props;
+
+  const handleFocus = () => {
+    setFocused(true);
+  };
+
+  return (
+    <div className='form__input'>
+      <label className='input__label'>{label}</label>
+      <input
+        className='input__field'
+        {...inputProps}
+        onChange={onChange}
+        onBlur={handleFocus}
+        // eslint-disable-next-line react/no-unknown-property
+        focused={focused.toString()}
+      />
+      {errorMessage && errorMessage !== '' && (
+        <span className='error__message'>{errorMessage}</span>
+      )}
+    </div>
+  );
+};
+
+FormInput.propTypes = {
+  label: PropTypes.string,
+  placeholder: PropTypes.string,
+  type: PropTypes.oneOf(['text', 'number', 'email']),
+  errorMessage: PropTypes.oneOf([
+    MESSAGES.FORM.USERNAME,
+    MESSAGES.FORM.AVATAR,
+    MESSAGES.FORM.EMAIL,
+    MESSAGES.FORM.PHONENUMBER,
+    MESSAGES.FORM.DESCRIPTION,
+    MESSAGES.FORM.ADDRESS,
+    MESSAGES.FORM.EMPTY,
+  ]),
+  pattern: PropTypes.oneOf([
+    REGEX.USERNAME,
+    REGEX.AVATAR,
+    REGEX.EMAIL,
+    REGEX.PHONENUMBER,
+    REGEX.DESCRIPTION,
+    REGEX.ADDRESS,
+  ]),
+};
+
+FormInput.defaultProps = {
+  label: 'Full name:',
+  type: 'text',
+  placeholder: 'Type name',
```

Comment:
what do you mean: `Type name`

Reply:
please consistent for placeholder. please pleaceholder will show for client if it's not clear. I make the user confuse

Reply:
I only let placeholder for fields having special characters
![image](https://github.com/datdragon777/React-Training/assets/74806826/9d03b87e-860d-45d9-82a8-08fe3ff64ae7)